src/library/abstractmodelroles.h

```diff
@@ -0,0 +1,17 @@
+#ifndef ABSTRACTMODELROLES_H
+#define ABSTRACTMODELROLES_H
+
+#include <QAbstractItemModel>
+
+enum AbstractRole {
```

Comment:
Since the AbstractRole is a continuation of the `Qt::ItemDataRole` and it's used in many places, it's assumed its `int` conversion and it cannot be changed to `enum class` without adding many castings to `int` and from `int` too, and it does not add any benefit in this case.